inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -171,7 +171,7 @@ def _rollback_and_delete_snap(self) -> None:
         a.) After reboot by SOTA, and diagnostic reports bad report for system health
         """
         logger.debug("")
-        if self.snap_num:
+        if self.snap_num and self.snap_num is not "0":
             self._dispatcher_broker.telemetry("SOTA attempting rollback")
             rc, err = self.trtl.sota_rollback(self.snap_num)
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Maintainability$</h2>**
**Suggestion:** Replace the identity check 'is not' with the equality operator '!=' when comparing strings.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        if self.snap_num and self.snap_num != "0":
            self._dispatcher_broker.telemetry("SOTA attempting rollback")
            rc, err = self.trtl.sota_rollback(self.snap_num)
```

Reply:
Agreed. Using  '!=' directly for string comparison is more idiomatic and clearer